src/main/kotlin/org/rust/lang/core/psi/ext/RsQualifiedNamedElement.kt

```diff
@@ -358,7 +299,16 @@ data class RsQualifiedName private constructor(
 
         private fun RsQualifiedNamedElement.toParentItem(): Item? {
             val name = ((this as? RsMod)?.modName ?: name) ?: return null
-            val itemType = when (this) {
+            if (this is RsMod && isCrateRoot) {
+                val crateName = containingCrate.asNotFake?.normName ?: return null
+                return Item(crateName, CRATE, this)
+            }
+            val itemType = parentItemType() ?: error("Unexpected type: `$this`")
```

Comment:
Let's try to avoid `error` here (I know that it was in original code written by me). Otherwise, it's quite simple to intorduce new `RsQualifiedNamedElement`, not to update this function and get error here in production since we also won't have tests for this new item

Let's better write error in log